SpiderBot.c

```diff
@@ -29,25 +24,36 @@ int main()
 // Setup the servo controller PCA9685 with Auto increment and a 50 hz freq.
 // @param eeBus The bus the controller is on.
 void setupServoController(const i2c *eeBus){
-  const char freq = 121; // set the freq register with 121 to get 50 hz (25Mhz / (4096* frq)) - 1.
-  const char sleep = 0x10;
-  const char autoInc = 0x20;
-  const char addressFrequency = 0xFE;
-  const char addressMode1 = 0x00;
-  char writeChar = 0;
+  const uint8_t controllerAddress = 0x40;
+  const uint8_t freq = 121; //(25Mhz / (4096* frq)) - 1.
+  const uint8_t sleep = 0x10;
+  const uint8_t autoInc = 0x20;
+  const uint8_t addressFrequency = 0xFE;
+  const uint8_t addressMode1 = 0x00;
+  uint8_t writeChar = 0;
   writeChar |= sleep;
-  i2c_out(eeBus, 0x40, addressMode1, 1, &writeChar, sizeof(writeChar));   // Enable sleep to set the freq register.
-  i2c_out(eeBus, 0x40, addressFrequency, 1, &freq, sizeof(freq));   // Set the freq to 50hz (20ms) for the servos.
+  i2c_out(eeBus, controllerAddress, addressMode1, 1, &writeChar, sizeof(writeChar));   // Enable sleep to set the freq register.
+  i2c_out(eeBus, controllerAddress, addressFrequency, 1, &freq, sizeof(freq));   // Set the freq for the servos.
   writeChar = 0 | autoInc;
-  i2c_out(eeBus, 0x40, addressMode1, 1, &writeChar, sizeof(writeChar)); // Set sleep off (default state) and enable auto increment.
+  i2c_out(eeBus, controllerAddress, addressMode1, 1, &writeChar, sizeof(writeChar)); // Set sleep off (default state) and enable auto increment.
 }
 
-void setServo(const i2c *eeBus, long start, long stop){
-  i2c_out(eeBus, 0x40, 0x06, 1, &start, sizeof(start));  // On pwm.
-  i2c_out(eeBus, 0x40, 0x08, 1, &stop, sizeof(stop));  // Off pwm.  
-}  
-
-void setServo(const i2c *eeBus, long start, long stop){
-  i2c_out(eeBus, 0x40, 0x06, 1, &start, sizeof(start));  // On pwm.
-  i2c_out(eeBus, 0x40, 0x08, 1, &stop, sizeof(stop));  // Off pwm.  
+// Set the servo from 0 to 180 degrees.
```

Comment:
Will add using the following style for doxygen:
/** Brief description which ends at this dot. Details follow
 *  here.
 */